File: C++/Sum of all substrings of a number.cpp
class Solution
{
    public:
    const int mod = 1000000007;
    long long sumSubstrings(string s){
        long long ans = 0, temp = 0;
        int n = s.size();
        for (int i = 0; i < n; i++){
            temp = temp * 10 + (i + 1) * (s[i] - '0');
            temp %= mod;
            ans += temp;
            ans %= mod;
        }
        return ans;
    }
};

File: JAVA/Sum of all substrings of a number.java
class Solution
{
    public static long sumSubstrings(String s)
    {
   	    int n = s.length();
        long mod = 1000000007;
        long result = 0;
        long multiplier = 1;
        for (int i = n - 1; i >= 0; i--) {
            long num = Long.parseLong(s.substring(i, i + 1)); 
            long contribution = (num * (i + 1) * multiplier) % mod; 
            result = (result + contribution) % mod; 
            multiplier = (multiplier * 10 + 1) % mod; 
        }
        return result;
    }
}
